src/raw_to_processed.py

```diff
@@ -0,0 +1,222 @@
+                "lpep_dropoff_datetime": "dropoff_datetime",
+                "ehail_fee": "fee"
+            },
+            inplace=True
+        )
+
+        # drop column
+        if "trip_type" in df.columns:
+            df.drop(columns=["trip_type"], inplace=True)
+
+    elif file.startswith("yellow"):
+        # rename columns
+        df.rename(
+            columns={
+                "tpep_pickup_datetime": "pickup_datetime",
+                "tpep_dropoff_datetime": "dropoff_datetime",
+                "airport_fee": "fee"
+            },
+            inplace=True
+        )
+
+    # fix data type in columns 'payment_type', 'dolocationid', 'pulocationid', 'vendorid'
+    if "payment_type" in df.columns:
+        df["payment_type"] = df["payment_type"].astype(int)
+    if "dolocationid" in df.columns:
+        df["dolocationid"] = df["dolocationid"].astype(int)
+    if "pulocationid" in df.columns:
+        df["pulocationid"] = df["pulocationid"].astype(int)
+    if "vendorid" in df.columns:
+        df["vendorid"] = df["vendorid"].astype(int)
+
+    # drop column 'fee'
+    if "fee" in df.columns:
+        df.drop(columns=["fee"], inplace=True)
+                
+    # Remove missing data
+    df = df.dropna()
+    df = df.reindex(sorted(df.columns), axis=1)
+    
+    print("Transformed data from file: " + file)
+
+    return df
+###############################################
+
+
+###############################################
+# Process data
+###############################################
+if __name__ == "__main__":
+
+    import s3fs
+
+    # Load minio config
+    cfg = load_cfg(CFG_FILE)
+    datalake_cfg = cfg["datalake"]
+    nyc_data_cfg = cfg["nyc_data"]
+
+    s3_fs = s3fs.S3FileSystem(
+        anon=False,
+        key=datalake_cfg["access_key"],
+        secret=datalake_cfg["secret_key"],
+        client_kwargs={'endpoint_url': "http://localhost:9000"}
+    )
+
+    # Create bucket 'processed'
+    create_bucket(datalake_cfg['bucket_name_2'])
+
+
+    for year in YEARS:
+
+        all_fps = glob(os.path.join(DATA_PATH, year, "*.parquet"))
+
+        for file in all_fps:
+            file_name = file.split('/')[-1]
+            print(f"Reading parquet file: {file_name}")
+
+            df = pd.read_parquet(file, engine='pyarrow')
+
+            # lower case all columns
+            df.columns = map(str.lower, df.columns)
+
+            df = drop_column(df, file_name)
+            df = merge_taxi_zone(df, file_name)
+            df = process(df, file_name)
+
+            # save to parquet file
+            path = f"s3://{datalake_cfg['bucket_name_2']}/{datalake_cfg['folder_name']}/" + file_name
+            df.to_parquet(path, index=False, filesystem=s3_fs, engine='pyarrow')
+            print("Finished transforming data in file: " + path)
+            print("==========================================================================================")
+###############################################
```

Comment:
Why do you use this?

Reply:
I want to read data from data/ (~raw) and push them (in parquet format) to the processed bucket. I cannot push data to Minio without any configuration to MinIO, so I use s3_fs.